todo_app/app.py

```diff
@@ -147,34 +193,61 @@ def add_todo():
                 to_add['due_date'] = datetime.strptime(date, '%d/%m/%Y').isoformat()
             
             db = DB()
-            db.add_item(MongoItem.from_dict(to_add, mode = "Save"))
+            todo_id = db.add_item(MongoItem.from_dict(to_add, mode = "Save"))
+            user: User = current_user
+            app.logger.info({ 
+                "category": LogCategory.todo, 
+                "action": LogAction.add_todo,
+                "user_id": user.id,
+                "user_role": user.role,
+                "todo_id": todo_id, 
```

Comment:
Nice, I can see you've carefully chosen to use id over the item description which is much safer from a sensitive-data perspective